includes/Checker/Checks/Trademarks_Check.php

```diff
@@ -0,0 +1,543 @@
+		'wh4tsapps',
+		'whatsapp',
+		'whats-app',
+		'watson',
+		'windows-',
+		'wocommerce',
+		'woocom-',
+		'woocommerce',  // technically ending with '-for-woocommerce' is allowed.
+		'woocomerce',
+		'woo-commerce',
+		'woo-',
+		'wo-',
+		'wordpress',
+		'wordpess',
+		'wpress',
+		'wp-',
+		'wp-mail-smtp-',
+		'yandex-',
+		'yahoo-',
+		'yoast',
+		'youtube-',
+		'you-tube-',
+	);
+
+	/**
+	 * Lists of trademarks that are allowed as 'for-whatever' ONLY.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @var string[]
+	 */
+	const FOR_USE_EXCEPTIONS = array(
+		'woocommerce',
+	);
+
+	/**
+	 * Lists of commonly used 'combo' names (to prevent things like 'woopress').
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @var string[]
+	 */
+	const PORTMANTEAUS = array(
+		'woo',
+	);
+
+	const TYPE_README = 1;
+	const TYPE_NAME   = 2;
+	const TYPE_SLUG   = 4;
+	const TYPE_ALL    = 7; // Same as all of the above with bitwise OR.
+
+	/**
+	 * Bitwise flags to control check behavior.
+	 *
+	 * @since n.e.x.t
+	 * @var int
+	 */
+	protected $flags = 0;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $flags Bitwise flags to control check behavior.
+	 */
+	public function __construct( $flags = self::TYPE_ALL ) {
+		$this->flags = $flags;
+	}
+
+	/**
+	 * Gets the categories for the check.
+	 *
+	 * Every check must have at least one category.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array The categories for the check.
+	 */
+	public function get_categories() {
+		return array( Check_Categories::CATEGORY_PLUGIN_REPO );
+	}
+
+	/**
+	 * Check for trademarks.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	protected function check_files( Check_Result $result, array $files ) {
+
+		// Check the trademarks in readme file plugin name.
+		if ( $this->flags & self::TYPE_README ) {
+			$this->check_for_name_in_readme( $result, $files );
+		}
+
+		// Check the trademarks in plugin name.
+		if ( $this->flags & self::TYPE_NAME ) {
+			$this->check_for_name_in_main_file( $result );
+		}
+
+		// Check the trademarks in plugin slug.
+		if ( $this->flags & self::TYPE_SLUG ) {
+			$this->check_for_slug( $result );
+		}
+	}
+
+	/**
+	 * Checks the trademarks in readme file plugin name.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	private function check_for_name_in_readme( Check_Result $result, array $files ) {
+		$plugin_relative_path = $result->plugin()->path();
+
+		// Find the readme file.
+		$readme_list = self::filter_files_by_regex( $files, '/readme\.(txt|md)$/i' );
+
+		// Filter the readme files located at root.
+		$potential_readme_files = array_filter(
+			$readme_list,
+			function ( $file ) use ( $plugin_relative_path ) {
+				$file = str_replace( $plugin_relative_path, '', $file );
+				if ( ! strpos( $file, '/' ) ) {
+					return true;
+				}
```

Comment:
A bit cleaner, we should also return `false` here. Also, maybe better to use `str_contains()`.
```suggestion
				return ! str_contains( $file, '/' );
```